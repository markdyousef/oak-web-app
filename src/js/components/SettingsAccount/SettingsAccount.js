import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';
import Input from '../shared/Input';
import Button from '../shared/Button';
import SavedIcon from '../../icons/saved';

const Container = styled.section`
    width: 100%;
    height: 100%;
    & h1 {
        margin-bottom: 20px;
    }
    & h5 {
        margin: 10px 5px;
        font-size: 14px;
        line-height: 17px;
        color: ${colors.grey}
    }
    & span {
        font-weight: bold;
    }
`;

const InputContainer = styled.div`
    margin-top: 20px;
`;

const ErrorBox = styled.div`
    height: 40px;
    color: ${colors.red}
`;

const ButtonContainer = styled.div`
    display: flex;
    justify-content: flex-end;
`;

const Saved = styled.div`
    color: ${colors.green};
    display: flex;
    align-items: center;
    font-weight: bold;
    font-size: 16px;
    & svg {
        fill: ${colors.green};
        height: 18;
        width: 18;
        margin-right: 5px;
    }
`;

export default class SettingsAccount extends Component {
    static propTypes = {
        updatePassword: PropTypes.func.isRequired
    }
    constructor() {
        super();
        this.state = {
            password: '',
            newPassword: '',
            confirmPassword: '',
            errorMessage: null,
            isSaved: false,
            errorField: null
        };
    }
    onSave = () => {
        const { password, newPassword, confirmPassword } = this.state;
        const { updatePassword } = this.props;

        if (confirmPassword !== newPassword) {
            this.setState({
                errorMessage: "New and confirmed password doesn't match",
                errorField: 'repeat'
            });
        }

        updatePassword(password, newPassword)
            .then(() => this.setState({ isSaved: true }))
            .catch((err) => console.log(err));
    }
    changeField = (key:String, value:String) => {
        switch (key) {
        case 'password':
            this.setState({ password: value });
            break;
        case 'newPassword':
            this.setState({ newPassword: value });
            break;
        case 'confirmPassword':
            this.setState({ confirmPassword: value });
            break;
        default:
            break;
        }
        this.setState({ errorMessage: null, isSaved: false, errorField: null });
    }
    render() {
        const { password, newPassword, confirmPassword, errorMessage, isSaved, errorField } = this.state;
        return (
            <Container>
                <h1>Account</h1>
                <InputContainer>
                    <Input
                        value={password}
                        type="password"
                        onChange={value => this.changeField('password', value)}
                        title="Password"
                    />
                    <h5>Enter your current password...</h5>
                </InputContainer>
                <InputContainer>
                    <Input
                        value={newPassword}
                        type="password"
                        onChange={value => this.changeField('newPassword', value)}
                        title="New password"
                        notValid={errorField === 'repeat'}
                    />
                    <h5>Create a strong new password to avoid trolls taking over your team ...</h5>
                </InputContainer>
                <InputContainer>
                    <Input
                        value={confirmPassword}
                        type="password"
                        onChange={value => this.changeField('confirmPassword', value)}
                        title="Confirm password"
                        notValid={errorField === 'repeat'}
                    />
                    <h5>... and confirm!</h5>
                </InputContainer>
                <ErrorBox>
                    {errorMessage}
                </ErrorBox>
                <ButtonContainer>
                    {(isSaved) ?
                        <Saved>
                            <SavedIcon />
                            Saved!
                        </Saved>
                        :
                        <Button
                            onClick={this.onSave}
                            text="Save"
                            type="primary"
                        />
                    }
                </ButtonContainer>
            </Container>
        );
    }
}
