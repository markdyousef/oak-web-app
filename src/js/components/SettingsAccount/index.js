import SettingsAccount from './SettingsAccount';

export default SettingsAccount;
