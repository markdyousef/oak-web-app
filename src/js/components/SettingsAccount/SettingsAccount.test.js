import React from 'react';
import { shallow } from 'enzyme';
import SettingsAccount from './SettingsAccount';

it('renders correctly', () => {
    const shallowComponent = shallow(<SettingsAccount />);
    expect(shallowComponent).toMatchSnapshot();
});
