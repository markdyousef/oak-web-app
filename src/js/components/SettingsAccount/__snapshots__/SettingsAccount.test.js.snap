// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.section>
    <h1>
        Account
    </h1>
    <styled.div>
        <Input
            notValid={false}
            onChange={[Function]}
            onKeyDown={null}
            placeholder={null}
            title="Password"
            type="password"
            value=""
        />
        <h5>
            Enter your current password.
        </h5>
    </styled.div>
    <styled.div>
        <Input
            notValid={false}
            onChange={[Function]}
            onKeyDown={null}
            placeholder={null}
            title="New password"
            type="password"
            value=""
        />
        <h5>
            Create a strong new password to avoid trolls taking over your team.
        </h5>
    </styled.div>
    <styled.div>
        <Input
            notValid={false}
            onChange={[Function]}
            onKeyDown={null}
            placeholder={null}
            title="Confirm password"
            type="password"
            value=""
        />
        <h5>
            Confirm your new password.
        </h5>
    </styled.div>
    <styled.div />
    <styled.div>
        <Unknown
            onClick={[Function]}
            text="Save"
            type="primary"
        />
    </styled.div>
</styled.section>,
  "nodes": Array [
    <styled.section>
      <h1>
            Account
      </h1>
      <styled.div>
            <Input
                  notValid={false}
                  onChange={[Function]}
                  onKeyDown={null}
                  placeholder={null}
                  title="Password"
                  type="password"
                  value=""
            />
            <h5>
                  Enter your current password.
            </h5>
      </styled.div>
      <styled.div>
            <Input
                  notValid={false}
                  onChange={[Function]}
                  onKeyDown={null}
                  placeholder={null}
                  title="New password"
                  type="password"
                  value=""
            />
            <h5>
                  Create a strong new password to avoid trolls taking over your team.
            </h5>
      </styled.div>
      <styled.div>
            <Input
                  notValid={false}
                  onChange={[Function]}
                  onKeyDown={null}
                  placeholder={null}
                  title="Confirm password"
                  type="password"
                  value=""
            />
            <h5>
                  Confirm your new password.
            </h5>
      </styled.div>
      <styled.div />
      <styled.div>
            <Unknown
                  onClick={[Function]}
                  text="Save"
                  type="primary"
            />
      </styled.div>
</styled.section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SettingsAccount
        updatePassword={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SettingsAccount {
        "_reactInternalInstance": [Circular],
        "changeField": [Function],
        "context": Object {},
        "onSave": [Function],
        "props": Object {
          "updatePassword": [Function],
        },
        "refs": Object {},
        "state": Object {
          "confirmPassword": "",
          "errorField": null,
          "errorMessage": null,
          "isSaved": false,
          "newPassword": "",
          "password": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.section>
          <h1>
                    Account
          </h1>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="Password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Enter your current password.
                    </h5>
          </styled.div>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="New password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Create a strong new password to avoid trolls taking over your team.
                    </h5>
          </styled.div>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="Confirm password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Confirm your new password.
                    </h5>
          </styled.div>
          <styled.div />
          <styled.div>
                    <Unknown
                              onClick={[Function]}
                              text="Save"
                              type="primary"
                    />
          </styled.div>
</styled.section>,
        "_debugID": 2,
        "_renderedOutput": <styled.section>
          <h1>
                    Account
          </h1>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="Password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Enter your current password.
                    </h5>
          </styled.div>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="New password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Create a strong new password to avoid trolls taking over your team.
                    </h5>
          </styled.div>
          <styled.div>
                    <Input
                              notValid={false}
                              onChange={[Function]}
                              onKeyDown={null}
                              placeholder={null}
                              title="Confirm password"
                              type="password"
                              value=""
                    />
                    <h5>
                              Confirm your new password.
                    </h5>
          </styled.div>
          <styled.div />
          <styled.div>
                    <Unknown
                              onClick={[Function]}
                              text="Save"
                              type="primary"
                    />
          </styled.div>
</styled.section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SettingsAccount
    updatePassword={[Function]}
/>,
}
`;
